lightningd/plugin.c

```diff
@@ -32,8 +49,124 @@ void plugin_register(struct plugins *plugins, const char* path TAKES)
 	p->cmd = tal_strdup(p, path);
 }
 
+/**
+ * Try to parse a complete message from the plugin's buffer.
+ *
+ * Internally calls the handler if it was able to fully parse a JSON message,
+ * and returns true in that case.
+ */
+static bool plugin_read_json_one(struct plugin *plugin)
+{
+	jsmntok_t *toks;
+	bool valid;
+	toks = json_parse_input(plugin->buffer, plugin->used, &valid);
```

Comment:
This works, but lightning-cli does it more efficiently, by saving the jsmn_parser and simply continuing when new data comes in.  Probably worth a note...